vendor/github.com/keybase/gomounts/linux.go

```diff
@@ -0,0 +1,45 @@
+// +build linux
+
+package gomounts
+
+/*
+#include <stdio.h>
+#include <stdlib.h>
+#include <mntent.h>
+*/
+import "C"
+import (
+	"errors"
+	"strings"
+	"unsafe"
+)
+
+func getMountedVolumes() ([]Volume, error) {
+	result := make([]Volume, 0)
+
+	cpath := C.CString("/proc/mounts")
+	defer C.free(unsafe.Pointer(cpath))
+	cmode := C.CString("r")
+	defer C.free(unsafe.Pointer(cmode))
+	var file *C.FILE = C.setmntent(cpath, cmode)
+	if file == nil {
+		return nil, errors.New("Unable to open /proc/mounts")
+	}
+	defer C.endmntent(file)
+	var ent *C.struct_mntent
+
+	mntopt := C.CString("user_id")
+	for ent = C.getmntent(file); ent != nil; ent = C.getmntent(file) {
+		mntType := C.GoString(ent.mnt_type)
+		uidstr := ""
+		if substr := C.GoString(C.hasmntopt(ent, mntopt)); len(substr) > 0 {
+			commasplit := strings.SplitN(substr, ",", 2)
+			equalsplit := strings.SplitN(commasplit[0], "=", 2)
```

Comment:
this seems a bit brittle to me, i'd have to test to see if it handles edge cases ok

Reply:
alternatively, we could just look at what `setmntopt` does and mimick its behavior.

Reply:
@akalin-keybase what is `setmntopt`?  Google shows nothing.

I think this code is pretty straightforward.  `hasmntopt` returns a substring starting at "user_id=", possibly with comma-separated options after it.  So we want the string between the first equal sign and the first comma.

Unless you have specific concerns, I'm not going to spend any more time on this part.

Reply:
sorry, meant `setmntent`

Reply:
I don't think `setmntent` does anything with options, it just opens a file: https://github.com/seL4/libmuslc/blob/master/src/linux/mntent.c#L8-L10

I'm going to leave this as is for now until we discover an issue with it.